scripts/build.sh

```diff
@@ -0,0 +1,214 @@
+            . + {
+                \"persistence\": {
+                    \"postgresql\": {
+                        \"secretRef\": {
+                            \"name\": \"sonataflow-psql-postgresql\",
+                            \"userKey\": \"postgres-username\",
+                            \"passwordKey\": \"postgres-password\"
+                        },
+                        \"serviceRef\": {
+                            \"name\": \"sonataflow-psql-postgresql\",
+                            \"port\": 5432,
+                            \"databaseName\": \"sonataflow\",
+                            \"databaseSchema\": \"${workflow_id}\"
+                        }
+                    }
+                }
+            }
+        )" "${sonataflow_cr}"
+        log_info "Added persistence configuration to the sonataflow CR"
+    fi
+}
+
+function build_image {
+    local image_name="${args["image"]%:*}"
+    local tag="${args["image"]#*:}"
+
+    # These add-ons enable the use of JDBC for persisting workflow states and correlation
+    # contexts in serverless workflow applications.
+    base_quarkus_extensions="\
+    org.kie:kie-addons-quarkus-persistence-jdbc:9.102.0.redhat-00005,\
+    io.quarkus:quarkus-jdbc-postgresql:3.8.6.redhat-00004,\
+    io.quarkus:quarkus-agroal:3.8.6.redhat-00004"
+
+    # The 'maxYamlCodePoints' parameter contols the maximum size for YAML input files. 
+    # Set to 35000000 characters which is ~33MB in UTF-8.  
+    base_maven_args_append="\
+    -DmaxYamlCodePoints=35000000 \
+    -Dkogito.persistence.type=jdbc \
+    -Dquarkus.datasource.db-kind=postgresql \
+    -Dkogito.persistence.proto.marshaller=false"
+    
+    if [[ -n "${QUARKUS_EXTENSIONS:-}" ]]; then
+        base_quarkus_extensions="${base_quarkus_extensions},${QUARKUS_EXTENSIONS}"
+    fi
+
+    if [[ -n "${MAVEN_ARGS_APPEND:-}" ]]; then
+        base_maven_args_append="${base_maven_args_append} ${MAVEN_ARGS_APPEND}"
+    fi
+
+    # Build specifically for linux/amd64 to ensure compatibility with OSL v1.35.0
+    pocker_args=(
+        -f="${args["workflow-directory"]}/docker/osl.Dockerfile"
+        --platform='linux/amd64'
+        --ulimit='nofile=4096:4096'
+        --build-arg="QUARKUS_EXTENSIONS=${base_quarkus_extensions}"
+        --build-arg="MAVEN_ARGS_APPEND=${base_maven_args_append}"
+    )
+    [[ -n "${args["builder-image"]:-}" ]] && pocker_args+=(--build-arg="BUILDER_IMAGE=${args["builder-image"]}")
+    [[ -n "${args["runtime-image"]:-}" ]] && pocker_args+=(--build-arg="RUNTIME_IMAGE=${args["runtime-image"]}")
+
+    log_info "Building the workflow image"
+    pocker build "${pocker_args[@]}" "${args["workflow-directory"]}"
+    pocker tag "${args["image"]}" "$image_name:$(git rev-parse --short=8 HEAD)"
+    if [[ "$tag" != "latest" ]]; then
+        pocker tag "${args["image"]}" "$image_name:latest"
+    fi
+    pocker images --filter="reference=$image_name" --format="{{.Repository}}:{{.Tag}}"
+}
+
+function maybe_deploy {
```

Comment:
Suggestion: split in 2: `push` and `deploy`

Reply:
I didn't do that because I didn't want to repeat the logic in the next 2 lines 😜